scipy/special/xsf/zeta.h

```diff
@@ -1,10 +1,371 @@
+	-120.60673920245685,
+	-124.28249333527555,
+	-127.95824746809424,
+	-131.63400160091294,
+	-135.30975573373163,
+	-138.9855098665503,
+	-142.661263999369,
+	-146.3370181321877,
+	-150.0127722650064,
+	-153.6885263978251,
+	-157.36428053064375,
+	-161.04003466346245,
+	-164.71578879628115,
+	-168.39154292909984,
+	-172.06729706191854,
+	-175.74305119473723,
+	-179.4188053275559,
+	-183.0945594603746
+    };
+
+    // Complex log of expansion coefficients for Euler-Maclaurin summation formula.
+    XSF_HOST_DEVICE inline std::complex<double> zeta_em_log_coeff(std::size_t n) {
+	std::complex<double> J(0.0, 1.0);
+	std::complex<double> result;
+	if (n < 50) {
+	    result = zeta_em_log_abs_coeff_lookup[n];
+	} else {
+	    /* Asymptotic formula
+	     * Uses https://dlmf.nist.gov/24.11#E1 to approximate B_{2n} and
+	     * Stirling's approximation for (2n)!.
+	     */
+	    result = std::log(2.0) - 2.0*n*std::log(2*M_PI);
+	}
+	if (n % 2 == 0) {
+	    /* B_{2n}/(2n)! is negative for even n. This contributes a term
+	     * pi*i when taking the log. */
+	    result += M_PI * J;
+	}
+	return result;
+    }
+
+    /* Compute riemann_zeta for complex input z using the Euler-Maclaurin formula.
+     * Computation of individual terms in expansion are logarithmized to avoid
+     * overflow. TODO: only logarithmize when necessary. */
+    XSF_HOST_DEVICE inline std::complex<double> zeta_euler_maclaurin(std::complex<double> z) {
+	if (z == 1.0) {
+	    /* Return NaN at pole since value depends on how z approaches 1.0. */
+	    return {std::numeric_limits<double>::quiet_NaN(), std::numeric_limits<double>::quiet_NaN()};
+	}
+	std::size_t n = static_cast<std::size_t>(std::max(std::abs(z.imag()) / 4.0, 50.0));
+	std::size_t m = n;
+	std::complex<double> result = 0.0;
+	for (std::size_t i = 1; i < n; i++) {
+	    std::complex<double> term = std::pow(static_cast<double>(i), -z);
+	    result += term;
+	    // When z.real() > 1, series converges and we can consider early termination
+	    if (z.real() > 1 && std::abs(term) / std::abs(result) <= std::numeric_limits<double>::epsilon()) {
+	     	return result;
+	    }
+	}
+	double N = static_cast<double>(n);
+	std::complex<double> b = std::pow(n, -z);
+	result += b * (0.5 + N / (z - 1.0));
+	std::complex<double> log_poch = std::log(z);
+	std::complex<double> log_factor = std::log(b) - std::log(N);
+	for (std::size_t i = 1; i <= m; i++) {
+	    std::complex<double> term = std::exp(zeta_em_log_coeff(i) + log_factor + log_poch);
+	    result += term;
+	    if (std::abs(term)/std::abs(result) <= std::numeric_limits<double>::epsilon()) {
+		return result;
+	    }
+	    log_poch += std::log(z + static_cast<double>(2*i - 1)) + std::log(z + static_cast<double>(2*i));
+	    log_factor -= 2*std::log(N);
+	}
+	// Euler-maclaurin absolute error estimate.
+	double error;
+	error = std::abs(std::exp(zeta_em_log_coeff(m + 1) + log_factor + log_poch));
+	error *= std::abs((z + 2.0*(m+1) + 1.0)/(z.real() + 2.0*(m+1) + 1.0));
```

Comment:
The sum is
```math
\sum_{k=1}^m T_{k, n}(s)
```
and your loop is `i = 1; i <= m`, so the last execution is with `i = m`.

The formula for the error is:
![{6A46A812-47FB-42DA-AAB5-FECEEED7D986}](https://github.com/user-attachments/assets/a4a8a9c0-af9f-4818-8a7e-e2570c39cd71)

So shouldn't this be:
```suggestion
	error *= std::abs((z + 2.0*m + 1.0)/(z.real() + 2.0*m + 1.0));
```